packages/actor-extract-links-extract-tree/lib/SolutionDomain.ts

```diff
@@ -0,0 +1,198 @@
+import { SolutionRange } from './SolutionRange';
+import { LogicOperator } from './solverInterfaces';
+
+const AN_EMPTY_SOLUTION_RANGE = new SolutionRange(undefined);
+/**
+ * A class representing the domain of a solution of system of boolean equation.
+ * Every operation return a new object.
+ */
+export class SolutionDomain {
```

Comment:
I would highly recommend making this class immutable, and producing new SolutionDomain's when doing operations on it.

Reply:
Ok good, but let's make all fields `readonly` then.